src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1049,21 +1057,45 @@ bool isRefEscape
             //  o no ref or out argument(excluding the receiver and arguments of ref-like types) may have a narrower ref-safe-to-escape than E1; and
             //  o   no argument(including the receiver) may have a narrower safe-to-escape than E1.
 
+            if (symbol.IsStatic)
```

Comment:
Do you have test for such static methods?

Reply:
Yes, the problem came up in some tests. Binder puts a fake receiver in static method calls - possibly for the purpose of semantic info. It looked as receiver was present and had to be counted in the analysis.

---
In reply to: [141156833](https://github.com/dotnet/roslyn/pull/22303#discussion_r141156833) [](ancestors = 141156833)

Reply:
For example tests like "RefLikeScopeEscapeThis" started hitting "Unreachable" crash because we would get unexpected "BiundTypeExpression" as a receiver. This change was needed to get behavior of static methods where it was before.

---
In reply to: [141160622](https://github.com/dotnet/roslyn/pull/22303#discussion_r141160622) [](ancestors = 141160622,141156833)